ripc-protocol-tcp/src/main/java/io/ripc/protocol/tcp/TcpInterceptor.java

```diff
@@ -0,0 +1,7 @@
+package io.ripc.protocol.tcp;
+
+public interface TcpInterceptor<I, O, II, OO> {
+
+    TcpHandler<II, OO> intercept(TcpHandler<I, O> handler);
```

Comment:
I think there's also the question of "what exactly are we intercepting" and "to what end?". Are we intercepting **only** data messages coming in and going out? Those are two different things, actually: reading requires a `subscribe` and writing an entirely separate `Publisher` with a possibly completely different type. In the TCP case they'll often be the same (ex: `ByteBuf`) but they likely won't be in request/response servers.

If we're intercepting for the purposes of providing, say authorization, then we're doing more than just message transformation because we would be possibly closing the connection (canceling the `Subscription`).

I'm wondering if "intercepting" shouldn't be done at the site of the event (`Subscription.request` and `onNext`) and as a translation/codec/whatever rather than incurring the overhead of wrapping, delegates, etc...